app/models/ems_refresh/save_inventory_network.rb

```diff
@@ -109,18 +110,18 @@ def save_network_groups_inventory(ems, hashes, target = nil)
     store_ids_for_new_records(ems.network_groups, hashes, :ems_ref)
   end
 
-  def save_cloud_subnets_inventory(network, hashes)
+  def save_cloud_subnets_inventory(network, hashes, _target = nil)
     hashes.each do |h|
-      %i(availability_zone parent_cloud_subnet).each do |relation|
+      %i(availability_zone parent_cloud_subnet cloud_network).each do |relation|
         h[relation] = h.fetch_path(relation, :_object) if h.fetch_path(relation, :_object)
       end
 
-      h[:ems_id] = network.ems_id
+      h[:ems_id] = network.ems_id if network.respond_to?(:ems_id)
     end
 
     save_inventory_multi(network.cloud_subnets, hashes, :use_association, [:ems_ref], nil, [:network_router])
 
-    network.save!
+    network.save! unless network.kind_of?(ManageIQ::Providers::NetworkManager)
```

Comment:
I don't understand this line...  what else would `network` be other than a `ManageIQ::Providers::NetworkManager`?